tensorflow/python/ops/parallel_for/control_flow_ops_test.py

```diff
@@ -2654,6 +2654,29 @@ def loop_fn(i):
 
     self._test_loop_fn(loop_fn, 4)
 
+
+  def test_nested_calls_loop_fn_autograph(self):
+
+    def inner(x):
+      for y in range(array_ops.constant(3)):
+        pass
+      return math_ops.square(x)
+
+    def outer(y):
+      return math_ops.reduce_sum(inner(y)) + 2
+
+    z = random_ops.random_uniform([4, 2])
+    
+    def loop_fn(i):
+      return outer(array_ops.gather(z, i))
+
+    @def_function.function
+    def loop_fn_caller():
+      self._test_loop_fn(loop_fn, 4)
```

Comment:
I don't think _test_loop_fn works inside tf.function.

Reply:
You probably just need to remove the decorator, and simplify loop_fn to just contain an autograph loop, no other inner/outer functions.

Reply:
I removed inner and outer that was just a copy of the test we had below.

> I don't think _test_loop_fn works inside tf.function.

Isn't exactly this what we want to fix and what was failing in the example we collected in the ticket https://github.com/tensorflow/tensorflow/issues/55340#issue-1177342161 ?

Reply:
I think it is, but `_test_loop_fn` itself contains test-only code like `self.evaluate`, that won't work the way you expect. So you might need to write a test in a way which doesn't use it, by manually calling whatever `_test_loop_fn` called.

Reply:
It might work though (accidentally). Let's see what the tests say.

Reply:
Ok tests are done. What do u think?

Reply:
Looking good, but I would add a TODO / caution in that test - I'm not sure if it works. If you still have the local build, try seeing if the test fails without the fix.

Reply:
> If you still have the local build,

I had tested it but now the cache is invalid.
I made an extra commit here to bring it back to its original behavior so you can see yourself abusing the CI a bit.

I have only a doubt about the codepath coverage as we still have two others conditional `loop_fn(` calls in the same file.

Reply:
Please check the test log, so I will revert the commit when you have done.

Reply:
You can check the log.
It Is failing as in ticket: `TypeError: 'Tensor' object cannot be interpreted as an integer`.
Let me know I will revert the commit

Reply:
Reverted. You can check the failing log at https://source.cloud.google.com/results/invocations/9e2f7686-e850-43e2-92eb-a3648f822ab4/targets/%2F%2Ftensorflow%2Fpython%2Fops%2Fparallel_for:control_flow_ops_test_cpu/tests